From: Mikulas Patocka <mpatocka@redhat.com>
Date: Sun, 19 Apr 2020 04:34:00 -0400
Subject: [PATCH] dm writecache: remove superfluous test in
 persistent_memory_claim
Git-commit: 499c18045eab16656ef4159c35b05865038f9f25
Patch-mainline: v5.8-rc1
References: bsc#1175995,jsc#SLE-15608

Remove superfluous test if dax_dev is NULL - dax_direct_access already
does this test.

Signed-off-by: Mikulas Patocka <mpatocka@redhat.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/md/dm-writecache.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/md/dm-writecache.c b/drivers/md/dm-writecache.c
index 613c171b1b6d..d29d3e234e01 100644
--- a/drivers/md/dm-writecache.c
+++ b/drivers/md/dm-writecache.c
@@ -234,10 +234,6 @@ static int persistent_memory_claim(struct dm_writecache *wc)
 
 	wc->memory_vmapped = false;
 
-	if (!wc->ssd_dev->dax_dev) {
-		r = -EOPNOTSUPP;
-		goto err1;
-	}
 	s = wc->memory_map_size;
 	p = s >> PAGE_SHIFT;
 	if (!p) {
-- 
2.16.4

